scripts/zones/Sacrarium/mobs/Old_Professor_Mariselle.lua

```diff
@@ -30,6 +116,41 @@ entity.onMobFight = function(mob, target)
             end
         end
     end
+
+    for i = OP_Mariselle+1, OP_Mariselle+2 do
+        local m = GetMobByID(i)
+        if m:isSpawned() then
+            m:updateEnmity(target)
+        end
+    end
+
+    local teleTime = mob:getLocalVar("teleTime")
+    if mob:getBattleTime() - teleTime > 30 and mob:getBattleTime() > 59 and mob:actionQueueEmpty() == true then
+        local profLocation = mob:getLocalVar("spawnLocation")
+        local randomPosition = math.random(1, 9)
+        utils.mobTeleport(mob, 2000, Locations[profLocation][randomPosition])
+
+        mob:setLocalVar("teleTime", mob:getBattleTime())
+    end
+
+    -- If players wander too far from professor and his teleport room he deaggros --
+    -- This is a safety measure due to the navmesh sucking, if players wanted too far and OPM + Babies are teleporting they will just wander through walls --
+    -- This happens to all mobs, but due to the teleport mechanics can sometimes cause issues --
+    --TO-DO Remove de-aggro when OOB Navmesh issues are fixed
```

Comment:
Space after `--`, also, can you change `TO-DO` to `TODO: `  Just trying to keep things consistent if people decide to search for something well, to do.